gordo_components/data_provider/ncs_reader.py

```diff
@@ -139,13 +144,14 @@ def read_tag_files(
                     sep=";",
                     header=None,
                     names=["Sensor", tag.name, "Timestamp", "Status"],
-                    usecols=[tag.name, "Timestamp"],
+                    usecols=[tag.name, "Timestamp", "Status"],
                     dtype={tag.name: np.float32},
                     parse_dates=["Timestamp"],
                     date_parser=lambda col: pd.to_datetime(col, utc=True),
                     index_col="Timestamp",
                 )
-
+                if remove_bad_values:
+                    df = df[df["Status"] != 0]
```

Comment:
And then here, it would be a one liner; `df = df[df["Status"].isin(filter_status_codes)]`